### PR TITLE
Refactors and Tguizes Orion Trail arcade machine

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

- [x] finish event datums
- [x] finish market
- [x] move orion into its own file, its large enough and i'm sick of all arcade machines being in one file

WOW all this for gbp well yeah here.

a few more things: now has an intructions page
lings before had literally no reason to kill people because the worst outcome of lings is that it kills someone anyway, now it's a game over if lings attack
few things have a confirmation prompt when they didn't

pioneers now get funny little moodies to reflect their situation and it goes up and down depending on how fucked they are. purely fluff!

## Why It's Good For The Game

tee goo ey - here's a sneak peek

image

image

## Changelog
:cl: tralezazelart
ui: tguized orion trail
code: Orion code maybe less shit?
refactor: Orion Events use datums whew
/:cl:

### CODE REVIEW

File: code/modules/tgui_panel/tgui_panel.dm [hunk 1]
```diff
@@ -48,12 +48,6 @@
 	))
 	window.send_asset(get_asset_datum(/datum/asset/simple/namespaced/fontawesome))
 	window.send_asset(get_asset_datum(/datum/asset/spritesheet/chat))
-	// Preload assets for /datum/tgui
-	var/datum/asset/asset_tgui = get_asset_datum(/datum/asset/simple/tgui)
-	var/flush_queue = asset_tgui.send(src.client)
-	if(flush_queue)
-		src.client.browse_queue_flush()
-	// Other setup
```
Comment (tralezab) [inline code/modules/tgui_panel/tgui_panel.dm:56]: This needs a different pr!

Comment (tralezab) [reply code/modules/tgui_panel/tgui_panel.dm:56]: #57649

File: code/game/machinery/computer/arcade/arcade.dm [hunk 2]
```diff
@@ -781,657 +763,234 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
 					if(2) //Food
 						if(food > 5)
 							fuel += 5
 							food -= 5
-							last_spaceport_action = "Traded Food for Fuel"
-							event()
-
-	add_fingerprint(usr)
+	add_fingerprint(gamer)
 	updateUsrDialog()
```
Comment (stylemistake) [inline code/game/machinery/computer/arcade/arcade.dm:932]: I don't think this is needed

Comment (tralezab) [reply code/game/machinery/computer/arcade/arcade.dm:932]: Good catch, remnant of the old system

**Comment by tralezab** [comment]
Last thing I need to fix before this is ready for testing/merge: I have some world messages hooked up to the targeted kill button because it's sending in the name of the target to kill but somehow isn't finding it in the execute crew proc

**Comment by tralezab** [comment]
damn these lints cray cray

File: code/game/machinery/computer/arcade/orion.dm [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+		Radio.set_frequency(FREQ_MEDICAL)
+		Radio.talk_into(src, "PSYCH ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please schedule psych evaluation.", FREQ_MEDICAL)
+
+		gamers[gamer] = -1
+
+		gamer.client.give_award(/datum/award/achievement/misc/gamer, gamer) // PSYCH REPORT NOTE: patient kept rambling about how they did it for an "achievement", recommend continued holding for observation
+		gamer.mind?.adjust_experience(/datum/skill/gaming, 50) // cheevos make u better
+
+		if(!isnull(GLOB.data_core.general))
+			for(var/datum/data/record/R in GLOB.data_core.general)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:133]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names

File: code/game/machinery/computer/arcade/orion.dm [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+		/datum/orion_event/raiders,
+		/datum/orion_event/illness,
+		/datum/orion_event/flux,
+		/datum/orion_event/black_hole,
+		/datum/orion_event/black_hole_death,
+		/datum/orion_event/changeling_infiltration,
+		/datum/orion_event/changeling_attack,
+		/datum/orion_event/space_port,
+		/datum/orion_event/space_port/tau_ceti,
+		/datum/orion_event/space_port_raid
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:40]:
```suggestion
		/datum/orion_event/space_port_raid,
```

trailing comma good

File: code/game/machinery/computer/arcade/orion.dm [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+	)
+	//actual amount of lings on board
+	var/lings_aboard = 0
+	//if the game should pretend there are lings on board.
+	var/lings_suspected = FALSE
+	var/spaceport_raided = FALSE
+	var/gameStatus = ORION_STATUS_START
+
+	var/obj/item/radio/Radio
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:49]:
```suggestion
	var/obj/item/radio/radio
```

File: code/game/machinery/computer/arcade/orion.dm [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+
+	if(gamers[gamer] > 2 && prob(20 * gamers[gamer]))
+
+		Radio.set_frequency(FREQ_SECURITY)
+		Radio.talk_into(src, "SECURITY ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please watch for violent behavior.", FREQ_SECURITY)
+
+		Radio.set_frequency(FREQ_MEDICAL)
+		Radio.talk_into(src, "PSYCH ALERT: Crewmember [gamer] recorded displaying antisocial tendencies in [get_area(src)]. Please schedule psych evaluation.", FREQ_MEDICAL)
+
+		gamers[gamer] = -1
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:127]: Move -1 to a define.

File: code/game/machinery/computer/arcade/orion.dm [hunk 7]
```diff
@@ -0,0 +1,524 @@
...
+		if("leave_spaceport")
+			gameStatus = ORION_STATUS_NORMAL
+			spaceport_raided = FALSE
+		if("raid_spaceport")
+			spaceport_raided = TRUE
+			encounter_event(/datum/orion_event/space_port_raid, gamer, gamerSkill, gamerSkillLevel, gamerSkillRands)
+		if("buyparts")
+			if(!spaceport_raided && fuel > 5)
+				switch(params["part"])
+					if(1) //Engine Parts
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:284]: Magic number, make this a define.

Comment (Mothblocks) [reply code/game/machinery/computer/arcade/orion.dm:284]: Everything else like it.

File: code/game/machinery/computer/arcade/orion.dm [hunk 8]
```diff
@@ -0,0 +1,524 @@
...
+				fuel += 7
+				food += 7
+		if("leave_spaceport")
+			gameStatus = ORION_STATUS_NORMAL
+			spaceport_raided = FALSE
+		if("raid_spaceport")
+			spaceport_raided = TRUE
+			encounter_event(/datum/orion_event/space_port_raid, gamer, gamerSkill, gamerSkillLevel, gamerSkillRands)
+		if("buyparts")
+			if(!spaceport_raided && fuel > 5)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:282]: Move the 5 out to a define, and use that in the buying too.

File: code/game/machinery/computer/arcade/orion.dm [hunk 9]
```diff
@@ -0,0 +1,524 @@
...
+			if(!spaceport_raided)
+				switch(params["what"])
+					if(1) //Fuel
+						if(fuel > 5)
+							fuel -= 5
+							food += 5
+					if(2) //Food
+						if(food > 5)
+							fuel += 5
+							food -= 5
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:301]: Magic numbeeeers

File: code/game/machinery/computer/arcade/orion.dm [hunk 10]
```diff
@@ -0,0 +1,524 @@
...
+	add_fingerprint(gamer)
+
+/**
+ * pickweights a new event, sets event var as it. it then preps the event if it needs it
+ *
+ * giving a path argument will instead find that instanced datum instead of pickweighting. Used in events that follow from events.
+ * Arguments:
+ * * path: if we want a specific event, this is the path of the wanted one
+ */
+/obj/machinery/computer/arcade/orion_trail/proc/encounter_event(path, gamer, gamerSkill, gamerSkillLevel, gamerSkillRands)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:311]: snake_case

File: code/game/machinery/computer/arcade/orion.dm [hunk 11]
```diff
@@ -0,0 +1,524 @@
...
+							fuel += 5
+							food -= 5
+	add_fingerprint(gamer)
+
+/**
+ * pickweights a new event, sets event var as it. it then preps the event if it needs it
+ *
+ * giving a path argument will instead find that instanced datum instead of pickweighting. Used in events that follow from events.
+ * Arguments:
+ * * path: if we want a specific event, this is the path of the wanted one
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:309]: There's a lot more arguments than this!

File: code/game/machinery/computer/arcade/orion.dm [hunk 12]
```diff
@@ -0,0 +1,524 @@
...
+				break
+	if(!event)
+		CRASH("Woah, hey! we could not find the specified event \"[path]\"! Add it to the events list, numb nuts!")
+	event.on_select(gamerSkill, gamerSkillLevel, gamerSkillRands)
+	if(obj_flags & EMAGGED)
+		event.emag_effect(gamer)
+
+/obj/machinery/computer/arcade/orion_trail/proc/lose_condition_met()
+
+		return TRUE
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:327]: what on earth is going on over here with this indentation

File: code/game/machinery/computer/arcade/orion.dm [hunk 13]
```diff
@@ -0,0 +1,524 @@
...
+		alive++
+	if(update)
+		new_settler_mood()//new faces!
+		update_static_data(usr)
+	return newcrew
+
+
+//Remove Random/Specific crewmember
+/obj/machinery/computer/arcade/orion_trail/proc/remove_crewmember(specific = "", dont_remove = "", update = TRUE)
+	var/list/safe2remove = settlers
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:383]: `safe_to_remove` ya dork

File: code/game/machinery/computer/arcade/orion.dm [hunk 14]
```diff
@@ -0,0 +1,524 @@
...
+	var/sheriff = remove_crewmember(target) //I shot the sheriff
+	if(target)
+		killed_crew += 1 //if there was no suspected lings, this is just plain murder
+	playsound(loc,'sound/weapons/gun/pistol/shot.ogg', 100, TRUE)
+	if(!settlers.len || !alive)
+		say("The last crewmember [sheriff], shot themselves, GAME OVER!")
+		if(obj_flags & EMAGGED)
+			gamer.death()
+		set_game_over(gamer, "Your last pioneer committed suicide.")
+		if(killed_crew >= 4)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:418]: This should use a define, or .len, or something to signify that its the number of crew you get.

File: code/game/machinery/computer/arcade/orion.dm [hunk 15]
```diff
@@ -0,0 +1,524 @@
...
+			supply_mood++
+		if(electronics)
+			supply_mood++
+		if(engine)
+			supply_mood++
+		supply_mood = min(supply_mood, 1) //they expect multiple things stocked
+		var/changing_mood = 0
+		if(prob(60)) //sometimes they just feel better or worse
+			changing_mood = rand(-1,1)
+		settlermoods[settlers[i]] +=  max(settlers.len + food_mood + supply_mood + changing_mood, 1)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:452]:
```suggestion
		settlermoods[settlers[i]] += max(settlers.len + food_mood + supply_mood + changing_mood, 1)
```

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 16]
```diff
@@ -0,0 +1,510 @@
...
+	///gaming skill level of the player
+	var/gamerSkillLevel = 0
+	///gaming skill of the player
+	var/gamerSkill = 0
+	///some other metric that makes it easier to do randoms with skill testing, god really 3 vars guys
+	var/gamerSkillRands = 0
+
+/datum/orion_event/New(_game)
+	. = ..()
+	game = _game
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:24]: This allows for named arguments. I like it.

```suggestion
/datum/orion_event/New(game)
	. = ..()
	src.game = game
```

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 17]
```diff
@@ -0,0 +1,510 @@
...
+	var/weight = 0
+	///buttons to pick in response to the event. Don't worry, orion js will handle the rest
+	var/list/event_responses = list()
+	///default emag effect of events is to play an audible message and sound
+	var/emag_message = "<span class='userdanger'>Coders forgot to set this!</span>"
+	var/emag_sound = 'sound/effects/bamf.ogg'
+	///our game
+	var/obj/machinery/computer/arcade/orion_trail/game
+	///gaming skill level of the player
+	var/gamerSkillLevel = 0
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:16]: snake_case

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 18]
```diff
@@ -0,0 +1,510 @@
...
+	var/text = "this displays a blurb about the event"
+	///pickweight to show up. will still be in the events pool if added to the events list but not RANDOM, only triggered.
+	var/weight = 0
+	///buttons to pick in response to the event. Don't worry, orion js will handle the rest
+	var/list/event_responses = list()
+	///default emag effect of events is to play an audible message and sound
+	var/emag_message = "<span class='userdanger'>Coders forgot to set this!</span>"
+	var/emag_sound = 'sound/effects/bamf.ogg'
+	///our game
+	var/obj/machinery/computer/arcade/orion_trail/game
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:14]: Weakref, or hook to its deletion to stop hard dels.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 19]
```diff
@@ -0,0 +1,510 @@
...
+/datum/orion_event/New(_game)
+	. = ..()
+	game = _game
+
+/**
+ * What happens when this event is selected to trigger, sets vars. also can set some event pre-encounter randomization
+ *
+ * Arguments:
+ * * _gamerSkill: gaming skill of the player
+ * * _gamerSkillLevel: gaming skill level of the player
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:31]: More of the `src.var = var` pattern here, please. Documentation with underscore-prefixed names is weird, and shows off the implementation detail.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 20]
```diff
@@ -0,0 +1,510 @@
...
+	..()
+	if(game.engine >= 1)
+		event_responses += "Fix Engine"
+	event_responses += "Wait"
+
+/datum/orion_event/engine_part/response(choice)
+	if(choice == "Fix Engine")
+		game.engine = max(0, --game.engine)
+	else
+		game.food -= ((game.alive + game.lings_aboard)*2)*3
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:84]: Magic numbers.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 21]
```diff
@@ -0,0 +1,510 @@
...
+	..()
+	if(game.electronics >= 1)
+		event_responses += "Repair Electronics"
+	event_responses += "Wait"
+
+/datum/orion_event/electronic_part/response(choice)
+	if(choice == "Repair Electronics")
+		game.electronics = max(0, --game.electronics)
+	else
+		game.food -= ((game.alive + game.lings_aboard)*2)*3
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:108]: Magic numbers here too.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 22]
```diff
@@ -0,0 +1,510 @@
...
+	You can replace the broken electronics with spares, \
+	or you can spend 3 days troubleshooting the AI."
+	weight = 2
+	//set by select
+	event_responses = list()
+
+/datum/orion_event/electronic_part/on_select(_gamerSkill, _gamerSkillLevel, gamerSkillRands)
+	..()
+	if(game.electronics >= 1)
+		event_responses += "Repair Electronics"
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:101]: I don't like the new-then-comparing strings. Can we make these defines as well? Prevents any possibility of typos and such.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 23]
```diff
@@ -0,0 +1,510 @@
...
+	event_responses += "Wait"
+
+/datum/orion_event/electronic_part/response(choice)
+	if(choice == "Repair Electronics")
+		game.electronics = max(0, --game.electronics)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:106]: Don't use the return of `++` or `--`, it's very confusing.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 24]
```diff
@@ -0,0 +1,510 @@
...
+
+/datum/orion_event/electronic_part/emag_effect(mob/living/carbon/gamer)
+	playsound(game, 'sound/effects/empulse.ogg', 50, TRUE)
+	game.visible_message("<span class='danger'>[src] malfunctions, randomizing in-game stats!</span>")
+	var/oldfood = game.food
+	var/oldfuel = game.fuel
+	game.food = rand(10,80) / rand(1,2)
+	game.fuel = rand(10,60) / rand(1,2)
+	if(game.electronics)
+		sleep(10)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:120]: `addtimer`.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 25]
```diff
@@ -0,0 +1,510 @@
...
+
+/datum/orion_event/hull_part/emag_effect(mob/living/carbon/gamer)
+	if(prob(10+gamerSkill))
+		game.say("Something slams into the floor around [src] - luckily, it didn't get through!")
+		playsound(game, 'sound/effects/bang.ogg', 50, TRUE)
+		return
+	playsound(game, 'sound/effects/bang.ogg', 100, TRUE)
+	var/turf/open/floor/F
+	for(F in orange(1, src))
+		F.ScrapeAway()
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:160]: What the fuck?

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 26]
```diff
@@ -0,0 +1,510 @@
...
+	game.say("Something slams into the floor around [src], exposing it to space!")
+	if(game.hull)
+		sleep(10)
+		game.say("A new floor suddenly appears around [src]. What the hell?")
+		playsound(game, 'sound/weapons/genhit.ogg', 100, TRUE)
+		var/turf/open/space/T
+		for(T in orange(1, src))
+			T.PlaceOnTop(/turf/open/floor/plating)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:168]: What the FUCK?

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 27]
```diff
@@ -0,0 +1,510 @@
...
+		text = " [ling1]'s arm twists and contorts into a grotesque blade!"
+
+	var/chance2attack = game.alive*20
+	if(!prob(chance2attack))
+		text += " [pick("Sensing unfavorable odds", "After a failed attack", "Suddenly breaking nerve")], \
+		the changeling[ling2 ? "s":""] vanish[ling2 ? "" : "es"] into space through the airlocks! You're safe... for now."
+		if(ling2)
+			game.lings_aboard = max(0,game.lings_aboard-2)
+		else
+			game.lings_aboard = max(0,--game.lings_aboard)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:354]: Don't use the result of `--`.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 28]
```diff
@@ -0,0 +1,510 @@
...
+			"[ling2 ? pick(ling1, ling2) : ling1] decapitates [deadguy] in a single cleaving arc!")
+		text += " [murder_text]"
+	else
+		text += " You valiantly fight off the changeling[ling2 ? "s":""]!"
+		if(ling2)
+			game.food += 30
+			game.lings_aboard = max(0, game.lings_aboard-2)
+		else
+			game.food += 15
+			game.lings_aboard = max(0, --game.lings_aboard)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:372]: Don't use the result of `--`.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 29]
```diff
@@ -0,0 +1,510 @@
...
+	name = "Looming Black Hole"
+	text = "Sensors indicate that a black hole's gravitational field is \
+	affecting the region of space we were headed through. We could stay \
+	of course, but risk of being overcome by its gravity, or we could \
+	change course to go around, which will take longer."
+	event_responses = list("Speed Past","Go Around")
+
+/datum/orion_event/black_hole/response(choice)
+	if(choice == "Go Around")
+		game.food -= ((game.alive + game.lings_aboard)*2)*3
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:387]: More magic numbers.

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 30]
```diff
@@ -0,0 +1,510 @@
...
+			if(!game.settlers.len)
+				game.set_game_over(usr, "You were gunned down by space port security.")
+				return ..()
+			text = "You failed to raid the spaceport! You lost [fuel*-1] Fuel and [food*-1] Food, AND [lost_crew] in your scramble to escape! ([fuel]fuel, [food] food, -Crew)"
+			if(game.obj_flags & EMAGGED)
+				game.say("WEEWOO! WEEWOO! Spaceport security en route!")
+				playsound(game, 'sound/items/weeoo1.ogg', 100, FALSE)
+				for(var/i, i<=3, i++)
+					var/mob/living/simple_animal/hostile/syndicate/ranged/smg/orion/O = new/mob/living/simple_animal/hostile/syndicate/ranged/smg/orion(get_turf(src))
+					O.target = usr
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:507]:
```suggestion
					var/mob/living/simple_animal/hostile/syndicate/ranged/smg/orion/orion_smg = new(get_turf(src))
					orion_smg.target = usr
```

File: code/modules/asset_cache/asset_list.dm [hunk 31]
```diff
@@ -185,6 +187,15 @@ GLOBAL_LIST_EMPTY(asset_datums)
 		sizes[size_id] = size = list(1, I, null)
 		sprites[sprite_name] = list(size_id, 0)
 
+/**
+ * A simple proc handing the Icon for you to modify before it gets turned into an asset.
+ *
+ * Arguments:
+ * * I: icon being turned into an asset
+ */
+/datum/asset/spritesheet/proc/ModifyInserted(icon/I)
```
Comment (Mothblocks) [inline code/modules/asset_cache/asset_list.dm:196]: Better variable names.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I have no idea what code you wrote and what you didn't, so it's time to do some editing-maybe-very-old-code. Too many magic numbers, etc.

File: code/game/machinery/computer/arcade/orion.dm [hunk 32]
```diff
@@ -0,0 +1,524 @@
...
+/obj/machinery/computer/arcade/orion_trail/ui_act(action, list/params)
+	. = ..()
+	if(.)
+		return
+	if(!iscarbon(usr))
+		return
+
+	. = TRUE
+
+	var/mob/living/carbon/gamer = usr
```
Comment (optimumtact) [inline code/game/machinery/computer/arcade/orion.dm:189]: I know you did the iscarbon check earlier but I think you should do this and then `if(!istype(gamer)) return`
instead as it means it's harder for someone to accidentally remove the iscarbon check without realising the cast exists and blowing shit the fuck up

File: code/game/machinery/computer/arcade/orion.dm [hunk 33]
```diff
@@ -0,0 +1,524 @@
...
+				reason = null
+				food = 80
+				fuel = 60
+				settlers = list("Harry","Larry","Bob")
+		if("continue")
+			if(turns >= ORION_TRAIL_WINTURN)
+				win(usr)
+				xp_gained += 34
+				return
+			usr?.mind?.adjust_experience(/datum/skill/gaming, xp_gained+1)
```
Comment (optimumtact) [inline code/game/machinery/computer/arcade/orion.dm:232]: use gamer here

Comment (optimumtact) [reply code/game/machinery/computer/arcade/orion.dm:232]: also stop abusing ?, you should only use that where it's actually sensible for the values to be potentially null.

**Comment by tralezab** [comment]
I need to handle some comments anturk made over discord before merge, mainly making the arcade event datums static

File: code/game/machinery/computer/arcade/orion.dm [hunk 34]
```diff
@@ -50,32 +42,48 @@
...
+	/datum/orion_event/flux,
+	/datum/orion_event/illness,
+	/datum/orion_event/engine_part,
+	/datum/orion_event/electronic_part,
+	/datum/orion_event/hull_part,
+	/datum/orion_event/space_port,
+	/datum/orion_event/space_port/tau_ceti,
+	/datum/orion_event/space_port_raid,
+	/datum/orion_event/black_hole,
+	/datum/orion_event/black_hole_death,
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion.dm:76]: indeeeent

File: code/game/machinery/computer/arcade/orion_event.dm [hunk 35]
```diff
@@ -0,0 +1,532 @@
...
+/datum/orion_event/black_hole_death/response(obj/machinery/computer/arcade/orion_trail/game, choice)
+	game.set_game_over(usr, "You were swept away into the black hole.")
+	..()
+
+/datum/orion_event/black_hole_death/emag_effect(obj/machinery/computer/arcade/orion_trail/game, mob/living/carbon/gamer)
+	if(game.obj_flags & EMAGGED)
+		playsound(game.loc, 'sound/effects/supermatter.ogg', 100, TRUE)
+		game.say("A miniature black hole suddenly appears in front of [game], devouring [gamer] alive!")
+		gamer.Stun(200, ignore_canstun = TRUE) //you can't run :^)
+		var/S = new /obj/singularity/academy(gamer.loc)
```
Comment (Mothblocks) [inline code/game/machinery/computer/arcade/orion_event.dm:440]: Better variable name.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Last one.

**Pull request merged**